src/binaryen-c.cpp

```diff
@@ -95,11 +100,13 @@ Literal fromBinaryenLiteral(BinaryenLiteral x) {
     case Type::v128:
       return Literal(x.v128);
     case Type::funcref:
-      return Literal::makeFuncref(x.func);
-    case Type::nullref:
-      return Literal::makeNullref();
+      return Literal::makeFunc(x.func);
     case Type::externref:
+    case Type::anyref:
+    case Type::eqref:
```

Comment:
Would it make more sense to have these types go to the `WASM_UNREACHABLE` until we have a more complete implementation that can handle values besides null?

Reply:
At most places, `anyref` etc. behave exactly like `externref` (which also can only be `null` as a Literal btw), so I didn't add separate `WASM_UNREACHABLE` cases there but put the cases at the places they'll ultimately end up at anyway. Would expect that once we'd want to represent other values than `null` for `externref` (and `anyref`, `eqref` by extension), we'd have to touch the exact sequence of cases anyway.